app/grandchallenge/cases/models.py

```diff
@@ -837,3 +840,111 @@ def save(self, *args, **kwargs):
                     kwargs={"post_process_image_task_pk": self.pk}
                 ).apply_async
             )
+
+
+class DICOMImageSetUploadStatusChoices(models.TextChoices):
+    INITIALIZED = "INITIALIZED", _("Initialized")
+    STARTED = "STARTED", _("Started")
+    FAILED = "FAILED", _("Failed")
+
+
+class DICOMImageSetUpload(UUIDModel):
+    creator = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        null=True,
+        default=None,
+        on_delete=models.SET_NULL,
+    )
+
+    user_uploads = models.ManyToManyField(
+        UserUpload, blank=True, related_name="dicom_import_jobs"
+    )
+
+    DICOMImageSetUploadStatusChoices = DICOMImageSetUploadStatusChoices
+    status = models.CharField(
+        max_length=11,
+        choices=DICOMImageSetUploadStatusChoices.choices,
+        default=DICOMImageSetUploadStatusChoices.INITIALIZED,
+        blank=False,
+    )
+
+    error_message = models.TextField(blank=True, default="")
+
+    class Meta:
+        verbose_name = "DICOM image set upload"
+        constraints = [
+            models.CheckConstraint(
+                check=models.Q(
+                    status__in=DICOMImageSetUploadStatusChoices.values
+                ),
+                name="dicomuimagesetupload_status_valid",
+            )
+        ]
+        indexes = (models.Index(fields=["status"]),)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__health_imaging_client = None
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
```

Comment:
Not sure about doing this here, `save` should just save as any `ValidationError` needs to be handled somehow, that is not possible here and that is different for everywhere the model is used (e.g. Forms directly or through a foreign key relationship, Celery Tasks, the admin etc). This is why it's not in `UUIDModel` or in Djangos core. For now, fine, but I think we just need to get in the habit of calling it or we might introduce new problems.